tests/unit/machine/all_machines_unittest.cc

```diff
@@ -0,0 +1,386 @@
+			y_mc[i] = (int32_t(y_mc[i]) + randi(prng)) % 3;
+		}
+	}
+
+	CDenseLabels* labels;
+	auto pt = machine->get_machine_problem_type();
+	switch (pt)
+	{
+	case PT_BINARY:
+	case PT_CLASS:
+	{
+		labels = new CBinaryLabels(y_binary);
+		break;
+	}
+
+	case PT_MULTICLASS:
+	{
+		labels = new CMulticlassLabels(y_mc);
+		break;
+	}
+
+	case PT_REGRESSION:
+		labels = new CRegressionLabels(y_reg);
+		break;
+
+	default:
+		SG_SERROR("Unsupported problem type: %d\n", pt);
+	}
+
+	SG_REF(features)
+	SG_REF(labels)
+	return make_pair(features, labels);
+}
+
+void serialize_machine(CMachine* machine, std::string& fname)
+{
+	auto fs = io::FileSystemRegistry::instance();
+	std::string class_name = machine->get_name();
+	fname = "shogun-unittest-AllMachines-trained_model_seiralization_consistency-" + class_name +
+			   ".XXXXXX";
+	generate_temp_filename(const_cast<char*>(fname.c_str()));
+
+	SG_REF(machine);
+	EXPECT_FALSE(fs->file_exists(fname));
+	std::unique_ptr<io::WritableFile> file;
+	EXPECT_FALSE(fs->new_writable_file(fname, &file));
+	auto fos = some<io::CFileOutputStream>(file.get());
+	auto serializer = some<io::CBitserySerializer>();
+	serializer->attach(fos);
+	serializer->write(wrap<CSGObject>(machine));
+}
+
+CMachine* deserialize_machine(std::string fname)
+{
+	auto fs = io::FileSystemRegistry::instance();
+	std::unique_ptr<io::RandomAccessFile> raf;
+	EXPECT_FALSE(fs->new_random_access_file(fname, &raf));
+	auto fis = some<io::CFileInputStream>(raf.get());
+	auto deserializer = some<io::CBitseryDeserializer>();
+	deserializer->attach(fis);
+	auto deser_obj = deserializer->read_object();
+	bool delete_success = !fs->delete_file(fname);
+	EXPECT_TRUE(delete_success);
+
+	return dynamic_cast<CMachine*>(deser_obj.get());
+}
+
+// TODO, generate this automatically, like in trained_model_serialization
+std::set<string> all_machines = {"LibSVM", "Perceptron", "LibLinear",
+		"MulticlassLibLinear", "LinearRidgeRegression", "KNN",
+		"KernelRidgeRegression", "LibLinearRegression", "RandomForest"};
+
+TEST(AllMachines, train_uninitialized)
+{
+	std::set<std::string> ignores = {
+				"RandomForest" // BaggingMachine.cpp line 153: Number of bag is not set!"
+				};
+		for (auto obj : sg_object_iterator<untemplated_sgobject>(all_machines).ignore(ignores))
+		{
+			auto machine = obj->as<CMachine>();
+			SCOPED_TRACE(machine->get_name());
+
+			EXPECT_THROW(machine->train(), ShogunException);
+		}
+}
+
+TEST(AllMachines, train_execute)
+{
+	std::set<std::string> ignores = {
+			"RandomForest" // BaggingMachine.cpp line 153: Number of bag is not set!"
+			};
+	for (auto obj : sg_object_iterator<untemplated_sgobject>(all_machines).ignore(ignores))
+	{
+		auto machine = obj->as<CMachine>();
+		SCOPED_TRACE(machine->get_name());
+
+		init_machine(machine);
+		auto data = generate_data(machine);
+		machine->set_labels(data.second);
+		machine->train(data.first);
+	}
+}
+
+TEST(AllMachines, train_thread_consistency)
+{
+	std::set<std::string> ignores = {
```

Comment:
copy-paste all around the source... plz avoid redundancy

Reply:
the ignores are test-specific.

Reply:
yeah 3x times the very same ignore

Reply:
this is because the number of tested machines is small and not yet automated. Although there are already differences here, see random forest. For more machines tested, there will be more diversity. If not, i can of course drop this.

Reply:
since you copy paste 3 times the value... you could create a value of it instead of copy-pasting the same string around. 1 copy-paste is one too many imo

Reply:
sure, i will look into this once the coverage is bigger